src/Sentry/Integrations/AppDomainProcessExitIntegration.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Sentry.Internal;
+
+namespace Sentry.Integrations
+{
+    internal class AppDomainProcessExitIntegration : IInternalSdkIntegration
+    {
+        private readonly IAppDomain _appDomain;
+        private IHub _hub;
+
+        public AppDomainProcessExitIntegration(IAppDomain appDomain = null)
+        {
+            _appDomain = appDomain ?? AppDomainAdapter.Instance;
+        }
+
+        public void Register(IHub hub, SentryOptions options)
+        {
+            Debug.Assert(hub != null);
+            _hub = hub;
+            _appDomain.ProcessExit += HandlerAsync;
+        }
+
+        public void Unregister(IHub hub)
+        {
+            _appDomain.ProcessExit -= HandlerAsync;
+            _hub = null;
+        }
+
+        private static readonly TimeSpan _flushTimeOut = TimeSpan.FromSeconds(1);
+
+        internal async void HandlerAsync(object sender, EventArgs e)
+        {
+            await FlushAsync(sender, e).ConfigureAwait(false);
```

Comment:
`async void` makes me scared. Should we `try/catch` suppress any errors here?

Reply:
Also, since it's not awaited, the process might exit before `FlushAsync` completes.
I believe the way to here is to go straight to `hub.Dispose()`.